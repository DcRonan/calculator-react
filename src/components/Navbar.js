import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav>
    <div>
      <div className="title"> Math Magicians </div>
    </div>
    <div className="navbar">
      <Link to="/">Home </Link>
      <Link to="/calculator">Calculator</Link>
      <Link to="/quote">Quote</Link>
    </div>
  </nav>
);

export default Navbar;
