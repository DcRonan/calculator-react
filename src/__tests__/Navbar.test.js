import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Navbar from '../components/Navbar';

describe('checks if Navbar component renders correctly', () => {
  test('should renders correct text', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    const text = screen.getByText('Math Magicians');
    expect(text).toBeInTheDocument();
  });

  test('should renders correct text', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    const text = screen.getByText('Quote');
    expect(text).toBeInTheDocument();
  });

  test('should render correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
