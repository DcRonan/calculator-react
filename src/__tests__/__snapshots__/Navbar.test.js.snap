// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Navbar component renders correctly should render correctly 1`] = `
<nav>
  <div>
    <div
      className="title"
    >
       Math Magicians 
    </div>
  </div>
  <div
    className="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home 
    </a>
    <a
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </div>
</nav>
`;
